sdk/communication/communication-common/test/identifierModelSerializer.spec.ts

```diff
@@ -0,0 +1,191 @@
+      },
+      {
+        kind: "communicationUser",
+        id: "8:acs:37691ec4-57fb-4c0f-ae31-32791610cb14_37691ec4-57fb-4c0f-ae31-32791610cb14"
+      }
+    );
+    assertSerialize(
+      { callingApplicationId: "28:37691ec4-57fb-4c0f-ae31-32791610cb14" },
+      { kind: "callingApplication", id: "28:37691ec4-57fb-4c0f-ae31-32791610cb14" }
+    );
+    assertSerialize(
+      { phoneNumber: "+1234555000" },
+      { kind: "phoneNumber", phoneNumber: "+1234555000" }
+    );
+    assertSerialize(
+      { microsoftTeamsUserId: "37691ec4-57fb-4c0f-ae31-32791610cb14", isAnonymous: false },
+      {
+        kind: "microsoftTeamsUser",
+        microsoftTeamsUserId: "37691ec4-57fb-4c0f-ae31-32791610cb14",
+        isAnonymous: false
+      }
+    );
+    assertSerialize(
+      { microsoftTeamsUserId: "37691ec4-57fb-4c0f-ae31-32791610cb14", isAnonymous: true },
+      {
+        kind: "microsoftTeamsUser",
+        microsoftTeamsUserId: "37691ec4-57fb-4c0f-ae31-32791610cb14",
+        isAnonymous: true
+      }
+    );
+    assertSerialize(
+      { id: "48:37691ec4-57fb-4c0f-ae31-32791610cb14" },
+      { kind: "unknown", id: "48:37691ec4-57fb-4c0f-ae31-32791610cb14" }
+    );
+  });
+
+  it("can deserialize", () => {
+    assertDeserialize(
+      {
+        kind: "communicationUser",
+        id: "8:acs:37691ec4-57fb-4c0f-ae31-32791610cb14_37691ec4-57fb-4c0f-ae31-32791610cb14"
+      },
+      {
+        kind: "CommunicationUser",
+        communicationUserId:
+          "8:acs:37691ec4-57fb-4c0f-ae31-32791610cb14_37691ec4-57fb-4c0f-ae31-32791610cb14"
+      }
+    );
+    assertDeserialize(
+      { kind: "callingApplication", id: "28:37691ec4-57fb-4c0f-ae31-32791610cb14" },
+      {
+        kind: "CallingApplication",
+        callingApplicationId: "28:37691ec4-57fb-4c0f-ae31-32791610cb14"
+      }
+    );
+    assertDeserialize(
+      { kind: "phoneNumber", phoneNumber: "+1234555000" },
+      { kind: "PhoneNumber", phoneNumber: "+1234555000" }
+    );
+    assertDeserialize(
+      {
+        kind: "microsoftTeamsUser",
+        microsoftTeamsUserId: "37691ec4-57fb-4c0f-ae31-32791610cb14",
+        isAnonymous: false
+      },
+      {
+        kind: "MicrosoftTeamsUser",
+        microsoftTeamsUserId: "37691ec4-57fb-4c0f-ae31-32791610cb14",
+        isAnonymous: false
+      }
+    );
+    assertDeserialize(
+      {
+        kind: "microsoftTeamsUser",
+        microsoftTeamsUserId: "37691ec4-57fb-4c0f-ae31-32791610cb14",
+        isAnonymous: true
+      },
+      {
+        kind: "MicrosoftTeamsUser",
+        microsoftTeamsUserId: "37691ec4-57fb-4c0f-ae31-32791610cb14",
+        isAnonymous: true
+      }
+    );
+    assertDeserialize(
+      { kind: "unknown", id: "48:37691ec4-57fb-4c0f-ae31-32791610cb14" },
+      { kind: "Unknown", id: "48:37691ec4-57fb-4c0f-ae31-32791610cb14" }
+    );
+  });
+
+  it("throws if kind not understood", () => {
+    assert.throws(() => {
```

Comment:
I thought we want this to return unknown identifier with the given id now that id is always the full MRI

Reply:
Right, I copied this over from the C# draft without thinking.